### ISSUE TITLE
Laravel artisan command make:job doesn't work as expected 

### ISSUE THREAD

**Issue by alpapan** [octobercms/docs#394]
Hello

When trying to user artisan the laravel way (for example make:job), it complaints 

```
  [RuntimeException]
  Unable to detect application namespace.
```

##### Reproduce steps

You install any official way and the result will be the same

```
curl -s https://octobercms.com/api/installer | php
grab composer.json test and server.php from github
composer install
```
or
```
git clone [...] (from master)
```
or
```
composer create-project october/october myoctober
```
or
```
composer create-project october/october myoctober-dev dev-master
```
**Any of the above and then:**
```
composer update
composer dumpautoload
composer diagnose
```

NB: i'm on cygwin windows, not that should make a difference...

This is not a fatal bug for me but it is a bug nonetheless because it breaks artisan so I'm reporting it... 

FWIW, there are no errant commas (the only reported solution on stackexchange) I can see in the composer.json (and diagnose/validate would report it anyway).

I finally managed to track it down though:

the composer.json needs these psr-4 configuration variables added (including the classmap from the october git).

```
    "autoload": {
        "psr-4": {
            "App\\": "app/"
        }
    },
    "autoload-dev": {
      "classmap": [
          "tests/TestCase.php",
          "tests/UiTestCase.php",
          "tests/PluginTestCase.php"
      ],
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
```

I would make a pull request if I knew how but the fix is pretty straightforward...

thank you....
alexie

**Comment by jjfmarket** [comment]
An appropriate error message would really help here

**Comment by LukeTowers** [comment]
@jjfmarket pull requests accepted! 😉

**Comment by marcogrueter** [comment]
How about omitting the registration of the unneccessary commands?
Since October\Rain\Foundation\Providers\ArtisanServiceProvider is extended from Illuminate\Foundation\Providers\ArtisanServiceProvider we could just override the $commands and $devCommands properties and remove the that aren't working.

**Comment by LukeTowers** [comment]
@marcogrueter not a bad idea, although I'd like if we could provide an error message telling the user what the proper command to use is

**Comment by marcogrueter** [comment]
@LukeTowers hmm that certainly would be nice, but I'm not sure how this could work.
There are only a few 'create' commands - yet - that are equivalent to a laravel native 'make' command and you'd have to intercept the call in command base class and kind of graylist the commands that have a proper use hint. But there are many commands that don't work at all - but this is out of scope for this issue.

It probably needs both:
Update to the docs with a "October does not support artisan make commands, please use the following scaffolding commands." hint and disabling them, along with commands like "session:table". Would also make for a short and sexy list on the command line.

**Comment by LukeTowers** [comment]
@marcogrueter that sounds reasonable to me. If October had full coverage for the incompatible artisan commands then I'd say a more specific message would be required but as of now just listing the October alternatives is probably good enough. Could you submit a PR?

**Comment by marcogrueter** [comment]
Sure thing, I'll submit one for the doc hint here and one for disabling the commands over at the library repo.

**Comment by ideepblue** [comment]
> Sure thing, I'll submit one for the doc hint here and one for disabling the commands over at the library repo.

Is there a way that adding back those Laravel artisan commands? I use `db:seed` a lot in my project. 

I tried to add `'Illuminate\Foundation\Providers\ArtisanServiceProvider',` in `config/app.php`, but it doesn't work.

```php
'providers' => array_merge(include(base_path('modules/system/providers.php')), [
        'System\ServiceProvider',
        'Illuminate\Foundation\Providers\ArtisanServiceProvider',
        'Jacobcyl\AliOSS\AliOssServiceProvider',
    ])
```

**Comment by LukeTowers** [comment]
@ideepblue what do you use `db:seed` for?

**Comment by ideepblue** [comment]
> @ideepblue what do you use `db:seed` for?

Hi @LukeTowers , we use it to run our seeder file from CLI. I knew I can do it through plugins updates, but it doesn't provide the flexibility as the `db:seed` command.

**Comment by ideepblue** [comment]
Anyway, I make it works by registering the `db:seed` in Plugin.php

```php
use Illuminate\Database\Console\Seeds\SeedCommand;

class Plugin extends PluginBase
    public function boot()
    {
        $this->app->singleton('command.seed', function ($app) {
            return new SeedCommand($app['db']);
        });
        $this->commands('command.seed');
    }
}
```

**Comment by LukeTowers** [comment]
@ideepblue fair enough, I typically just utilize the plugin update system and `php artisan october:up` but manually specifying the seeder class to be used is a valid use case. @bennothommo could you re-add it?

**Comment by bennothommo** [comment]
@LukeTowers @ideepblue `db:seed` is now re-added: https://github.com/octobercms/library/commit/9bc02271fdb74c3326d977549062f659e7f24095
